notion-core/src/main/scala/com/ambiata/notion/core/Location.scala

```diff
@@ -91,6 +91,41 @@ object Location {
       e.getMessage.left
   }
 
+  def localLocationFromUri(s: String): String \/ Location =
+    try {
+      val uri = new java.net.URI(s)
+      uri.getScheme match {
+        case "file" =>
+          LocalLocation(uri.toURL.getFile).right
+        case null =>
+          LocalLocation(uri.getPath).right
+        case _ =>
+          s"Not a local location [${uri.getScheme}]".left
+      }
+    } catch { case e: java.net.URISyntaxException => e.getMessage.left }
+
+  def s3LocationFromUri(s: String): String \/ Location =
+    try {
+      val uri = new java.net.URI(s)
+      uri.getScheme match {
+        case "s3" =>
+          S3Location(uri.getHost, uri.getPath.drop(1)).right
```

Comment:
The URI `s3:/bucket/key` will mean `scheme` will be `s3` but `getHost` will return `null`